ui/components/data-sources/options/PostgresOptions.js

```diff
@@ -1,83 +1,141 @@
-            <Col sm={9}>
-                <FormControl
-                    disabled={isDisabled("url")}
-                    type="text"
-                    value={options.url}
-                    onChange={e => onOptionsChange({
-                        ...options,
-                        url: e.target.value
-                    })}
-                />
-            </Col>
-        </FormGroup>
-        <FormGroup controlId="port" validationState={validations.port}>
-            <Col sm={3}>Port</Col>
-            <Col sm={9}>
-                <FormControl
-                    disabled={isDisabled("port")}
-                    type="text"
-                    value={options.port}
-                    onChange={e => onOptionsChange({
-                        ...options,
-                        port: e.target.value
-                    })}
-                />
-            </Col>
-        </FormGroup>
-        <FormGroup controlId="database" validationState={validations.database}>
-            <Col sm={3}>Database</Col>
-            <Col sm={9}>
-                <FormControl
-                    disabled={isDisabled("database")}
-                    type="text"
-                    value={options.database}
-                    onChange={e => onOptionsChange({
-                        ...options,
-                        database: e.target.value
-                    })}
-                />
-            </Col>
-        </FormGroup>
-        <FormGroup controlId="username" validationState={validations.username}>
-            <Col sm={3}>Username</Col>
-            <Col sm={9}>
-                <FormControl
-                    disabled={isDisabled("username")}
-                    type="text"
-                    value={options.username}
-                    onChange={e => onOptionsChange({
-                        ...options,
-                        username: e.target.value
-                    })}
-                />
-            </Col>
-        </FormGroup>
-        <FormGroup controlId="password" validationState={validations.password}>
-            <Col sm={3}>Password</Col>
-            <Col sm={9}>
-                <FormControl
-                    disabled={isDisabled("password")}
-                    type="password"
-                    value={options.password}
-                    onChange={e => onOptionsChange({
-                        ...options,
-                        password: e.target.value
-                    })}
-                />
-            </Col>
-        </FormGroup>
-    </React.Fragment>
-);
+import { Validate, Validators } from "../../../helper/Validators";
+
+const INITIAL_STATE = {
+    urlValidation: "null",
+    portValidation: "success",
+    databaseValidation: "null",
+    usernameValidation: "null",
+    passwordValidation: "success"
+};
+
+class PostgresOptions extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = INITIAL_STATE;
+    }
+
+    onUrlChange(url) {
```

Comment:
Why not one `onChange` function that handles all validations? Bit less performant but fewer lines of code.